clients/amqp-connection/src/main/java/org/eclipse/hono/client/amqp/connection/AmqpConstants.java

```diff
@@ -93,6 +93,16 @@ public Authorities getAuthorities() {
         }
     };
 
+    /**
+     * The name of the boolean system property with which the usage of the legacy trace context format can be enabled.
+     */
+    public static final String SYSTEM_PROPERTY_USE_LEGACY_TRACE_CONTEXT_FORMAT = "hono.tracing.amqp.useLegacyTraceContextFormat";
+
+    /**
+     * Value of the {@link #SYSTEM_PROPERTY_USE_LEGACY_TRACE_CONTEXT_FORMAT} system property.
+     * Determines whether the legacy trace context format shall be used.
+     */
+    public static boolean USE_LEGACY_TRACE_CONTEXT_FORMAT = Boolean.getBoolean(SYSTEM_PROPERTY_USE_LEGACY_TRACE_CONTEXT_FORMAT);
```

Comment:
this means that by default, the legacy context format is **not** used, right? FMPOV this could break existing system set-ups so I would rather default to the legacy format and allowing users to explicitly enable the format defined by the standard. With Hono 3.0 we can then turn it around, i.e. use the standard format by default and allow users to explicitly enable the legacy format.

Reply:
Ok, we can stick to the legacy format as default.
In order to make it easier to set the new format, I would switch to using a corresponding AMQP client config property then, instead of the system property. Having to set the system property by overriding the `javaOptions` Hono Helm chart config value would be rather inconvenient (having to copy the `-XX:MinRAMPercentage=80 -XX:MaxRAMPercentage=80` `javaOptions` default value).

Reply:
There is now a `${prefix}.useLegacyTraceContextFormat` AMQP client connection property (default is `true`).
This applies to the places in the code where the trace context get injected into an AMQP message.

As the AMQP client connection configuration doesn't apply in the places in the code, where a trace context gets extracted, I've chosen a configuration-independent approach there. If the legacy `x-opt-trace-context` message annotation exists, the trace-context is extracted from it, otherwise the application properties are taken.